Add compatibility with lite-xl

prantlf opened the pull request:
Thank you for your work on this! If you are interested in the compatibility with `lite-xl`, this is the minimum change that I needed.

I dared to change `EXEDIR` for `USERDIR`. Were you really planning to manage plugins in the installation directory and not the plugins in your `$HOME`?

prantlf committed:
Support Lite XL, List local plugins from USERDIR

prantlf committed:
Introduce a function to recognise lite-xl

Use the original rxi/lite-plugins in lite.

takase1121 commented:
> I dared to change EXEDIR for USERDIR. Were you really planning to manage plugins in the installation directory and not the plugins in your $HOME?

Of course I don't dare to, but there's no such things as `USERDIR` in lite :)

takase1121 commented on init.lua:
```diff
@@ -7,11 +8,15 @@ local style = require "core.style"
-local PLUGIN_PATH = system.absolute_path(EXEDIR .. "/data/plugins")
-local PLUGIN_URL = "https://raw.githubusercontent.com/rxi/lite-plugins/master/README.md"
-
+local PLUGIN_PATH = system.absolute_path(USERDIR .. "/plugins")
```
plugins are installed in `EXEDIR` in lite. Consider wrapping it in a conditional.

prantlf replied:
Oh, of course. It did not occur to me.

prantlf committed:
Make usage of USERDIR specific to lite-xl, otherwise use EXEDIR

Users of the original Lite will have to make the plugin directory in the application
installation directory writable, to be able to use this plugin.

prantlf commented:
> > I dared to change EXEDIR for USERDIR. Were you really planning to manage plugins in the installation directory and not the plugins in your $HOME?
> 
> Of course I don't dare to, but there's no such things as `USERDIR` in lite :)

Oh, I did not expect it. How do the users of the original Lite handle upgrades? If they edit a plugin bundled with the application, their changes will be overwritten.

Another problem could be an upgrade or reinstallation of Lite. For example, Homebrew purges the installation directory during the upgrade, which would remove all plugin customisations from EXEDIR.

Initially I thought that the users of the original Lite has to make the `plugins` directory writable to be able to use your plugin, which would be ugly, but it appears, that they might not have to. Permissions in my installation directory:

    $ ls -al /Applications/lite-xl.app/Contents/Resources/
    total 68
    drwxr-xr-x  7 prantlf staff   224 Mai 28 16:37 ./
    drwxr-xr-x  6 prantlf staff   192 Mai 28 16:37 ../
    drwxr-xr-x 22 prantlf staff   704 Mai 28 16:37 colors/
    drwxr-xr-x 26 prantlf staff   832 Mai 28 16:35 core/
    drwxr-xr-x  5 prantlf staff   160 Mai 28 16:35 fonts/
    -rw-r--r--  1 prantlf staff 68691 Mai 28 16:37 icon.icns
    drwxr-xr-x 22 prantlf staff   704 Mai 28 16:35 plugins/

Probably because I installed the application myself and I am an admin of the machine, I can modify it. I do not know if other machines look the same, but now I understand why developers did not need the USERDIR from the beginning.

The usage of USERDIR instead of EXEDIR is something that I did not intend to be in the Lite XL condition. 
The compatibility should not change the plugin behaviour and cause bad surprises. I wonder if USERDIR could be introduced in the original Lite too?

takase1121 commented:
> Oh, I did not expect it. How do the users of the original Lite handle upgrades? If they edit a plugin bundled with the application, their changes will be overwritten.

You're thinking too much. You don't "install" lite as a typical unix like program, you just unzip and use it.

There's no way rxi will introduce USERDIR. First, you have XDG defining multiple paths for things, Windows has its own minds and MacOS' occasional bullshit, I don't see a reason for rxi to do it.

prantlf commented:
> > Oh, I did not expect it. How do the users of the original Lite handle upgrades? If they edit a plugin bundled with the application, their changes will be overwritten.
> 
> You're thinking too much. You don't "install" lite as a typical unix like program, you just unzip and use it.

Well, that may not last long. With the growing user base and maturing of the application, just unzipping will not do. If you were maintaining all your development tools like this, you would not be doing anything else...

I use Homebrew to maintain all my tools like the UNIX dev tools, including lite-xl.

> There's no way rxi will introduce USERDIR. First, you have XDG defining multiple paths for things, Windows has its own minds and MacOS' occasional bullshit, I don't see a reason for rxi to do it.

Well, that is too bad. An application that is supposed to be customised cannot grow out of being a private tool in the build directory. I see no problem with handling the three different user data directories, while the rest of the code is shared.

takase1121 commented:
You can try to convince rxi :) Last time franko did it it was kinda awkward
in the end.

I use a few suckless programs so doing something similiar didn't really bother me anymore. 

On Mon, Jun 7, 2021, 15:13 Ferdinand Prantl ***@***.***>
wrote:

> Oh, I did not expect it. How do the users of the original Lite handle
> upgrades? If they edit a plugin bundled with the application, their changes
> will be overwritten.
>
> You're thinking too much. You don't "install" lite as a typical unix like
> program, you just unzip and use it.
>
> Well, that may not last long. With the growing user base and maturing of
> the application, just unzipping will not do. If you were maintaining all
> your development tools like this, you would not be doing anything else...
>
> I use Homebrew to maintain all my tools like the UNIX dev tools, including
> lite-xl.
>
> There's no way rxi will introduce USERDIR. First, you have XDG defining
> multiple paths for things, Windows has its own minds and MacOS' occasional
> bullshit, I don't see a reason for rxi to do it.
>
> Well, that is too bad. An application that is supposed to be customised
> cannot grow out of being a private tool in the build directory. I see no
> problem with handling the three different user data directories, while the
> rest of the code is shared.
>
> —
> You are receiving this because you modified the open/close state.
> Reply to this email directly, view it on GitHub
> <https://github.com/takase1121/lite-plugin-manager/pull/1#issuecomment-855659265>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AE6UHTHIE25OLRDPROZOUGLTRRWSDANCNFSM46GFD3QA>
> .
>

prantlf commented:
> You can try to convince rxi :) Last time franko did it it was kinda awkward in the end. I use a few suckless programs so doing something similiar didn't really bother me anymore.

I will try it then :-) One 1.16.12 is out, I will prepare a package for homebrew-cask which will demonstrate the use case better.
